pcmanfm/application.cpp

```diff
@@ -272,7 +272,7 @@ bool Application::parseCommandLineArgs() {
     }
     else {
         QDBusConnection dbus = QDBusConnection::sessionBus();
-        QDBusInterface iface(QLatin1String(serviceName), QStringLiteral("/Application"), QLatin1String(ifaceName), dbus, this);
+        QDBusInterface iface(QLatin1String(serviceName), QStringLiteral("/org/freedesktop/FileManager1"), QLatin1String(ifaceName), dbus, this);
```

Comment:
This doesn't fix the main problem. If another file manager also registers `/org/freedesktop/FileManager1` (as dolphin does), pcmanfm-qt can't run after it. Of course, we don't want that; pcmanfm-qt should be usable everywhere.

Reply:
You need to

 1. Restore "org.pcmanfm.PCManFM", "org.pcmanfm.Application", and "/Application" in `application.cpp`;
 2. Restore `<interface name="org.pcmanfm.Application">` in `pcmanfm/org.pcmanfm.Application.xml`; and
 3. Register the service "org.freedesktop.FileManager1" and the object "/org/freedesktop/FileManager1" in `application.cpp`.

Then, if pcmanfm-qt is started before Dolphin, it'll control "org.freedesktop.FileManager1". If Dolphin is already started, pcmanfm-qt will start without problem but "org.freedesktop.FileManager1" will be controlled by Dolphin but we don't expect anything else.

Of course, I mentioned Dolphin just as an example.

Reply:
@tsujan, @ahsand97 since this is probably relevant for Filer too I would like to learn about how to do this "the right way". I never quite understood why we need XML files when we can expose services over D-Bus. Isn't that kind of defeating the purpose of publishing stuff on D-Bus?

Since you apparently understand how this all works, can you point me to good documentation to learn about why XML files are needed here?

> If another file manager also registers /org/freedesktop/FileManager1 (as dolphin does)

Wouldn't we want whatever file manager is running to react to those D-Bus messages? E.g., if the user is running a KDE session with Dolphin, then clicking on "Show in file manager" in the Downloads window of a web browser should show the downloaded file in Dolphin. But on Lxqt where the user is running PcmanFM-Qt as part of the session, clicking on "Show in file manager" in the Downloads window of a web browser should show the downloaded file in PcmanFM-Qt, no? To achieve this, I was assuming that the first file manager that tries to register `/org/freedesktop/FileManager1` "wins". But maybe this assumption was wong?

Reply:
> Wouldn't we want whatever file manager is running to react to those D-Bus messages?

We do but we don't want that file manager to prevent pcmanfm-qt from running. Therefore, we need to register `org.freedesktop.FileManager1` only for that purpose. Previously, in this PR, it was also registered for providing a single-instance of the app, while `org.pcmanfm.PCManFM` should have been used.

Reply:
Thanks, got it.